src/compiler/transformers/classFields.ts

```diff
@@ -759,7 +770,11 @@ namespace ts {
                 // Declare private names.
                 for (const member of node.members) {
                     if (isPrivateIdentifierClassElementDeclaration(member)) {
-                        addPrivateIdentifierToEnvironment(member);
+                        const error = addPrivateIdentifierToEnvironment(member);
+                        if (error) {
+                            // leave invalid code as it is
+                            return node.members;
+                        }

```

Comment:
discussion: I'm a bit worries here about the decision to bail on all member transforms. If a pair of members is invalid, we can skip the transform for them. But it would be useful for debugging to only skip transform for those. I see emit as best effort, and this seems to bail a bit to quickly for my taste